textile/features.textile

```diff
@@ -688,7 +688,9 @@ h3(#realtime-channel). RealtimeChannel
 ** @(RTL7b)@ Subscribe with a name argument and a listener argument subscribes a listener to only messages whose @name@ member matches the string name
 ** @(RTL7c)@ This clause has been replaced by "@RTL7g@":#RTL7g. It was valid up to and including specification version @3.0@.
 ** @(RTL7g)@ If the @attachOnSubscribe@ channel option is @true@, implicitly attaches the @RealtimeChannel@ if the channel is in the @INITIALIZED@, @DETACHING@, or @DETACHED@ states. The optional callback, if provided, is called according to "@RTL4d@":#RTL4d based on the implicit attach operation. The listener will always be registered regardless of the implicit attach result
-** @(RTL7h)@ If the @attachOnSubscribe@ channel option is @false@, the optional callback must not be called.
+** @(RTL7h)@ If the @attachOnSubscribe@ channel option is @false@, then the behaviour depends on the public API that a given SDK uses to communicate the result of an "@RTL7g@":#RTL7g implicit attach:
+*** If the SDK’s API accepts an optional callback to communicate the result of an "@RTL7g@":#RTL7g implicit attach, then it is a programmer error to provide such a callback when the @attachOnSubscribe@ channel option is @false@. This programmer error should be handled in an idiomatic fashion (e.g. by halting the program).
```

Comment:
```suggestion
*** If the SDK’s API accepts an optional callback to communicate the result of an "@RTL7g@":#RTL7g implicit attach, then it is a programmer error to provide such a callback when the @attachOnSubscribe@ channel option is @false@. This programmer error should be handled in an idiomatic fashion (e.g. throwing exception or returning error without callback being called).
```

Reply:
by halting program do we mean exiting the application? like calling `exit(0)` or something

Reply:
Yeah, that's roughly what Swift does — programmer errors are considered to be irrecoverable. That's why I said "in an idiomatic fashion".

How would, say, Java or Go want to handle this? Would they throw an `ErrorInfo`? If so, do you have any thoughts on what would be an appropriate `statusCode` and `code` to assign to this error, so that I can specify it here? (Deciding what codes to use is something that remains a bit of a mystery to me.)

Reply:
I think throwing exception with common error code should be enough, as long as exception message convey the purpose

Reply:
I’ve added this info.